pkg/provision/cleanup.go

```diff
@@ -0,0 +1,178 @@
+package provision
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/tfexplorer/client"
+	"github.com/threefoldtech/tfexplorer/models/generated/workloads"
+	"github.com/threefoldtech/zbus"
+	"github.com/threefoldtech/zos/pkg/app"
+	"github.com/threefoldtech/zos/pkg/provision/common"
+	"github.com/threefoldtech/zos/pkg/storage"
+	"github.com/threefoldtech/zos/pkg/stubs"
+	"github.com/threefoldtech/zos/pkg/zdb"
+	"golang.org/x/net/context"
+)
+
+// CleanupResources cleans up unused resources
+func CleanupResources(ctx context.Context, zbus zbus.Client) error {
+	explorer, err := app.ExplorerClient()
+	if err != nil {
+		return err
+	}
+	storaged := stubs.NewStorageModuleStub(zbus)
+
+	toSave, err := checkContainers(ctx, zbus)
+	if err != nil {
+		return errors.Wrap(err, "failed to check containers")
+	}
+
+	fss, err := storaged.ListFilesystems()
+	if err != nil {
+		return err
+	}
+
+	for _, fs := range fss {
+		log.Info().Msgf("checking subvol %s", fs.Path)
+		// Don't delete zos-cache!
+		if fs.Name == storage.CacheLabel {
```

Comment:
I don't really know how the cache is treated right now in other modules. But seeing this I wonder if it would not be better to treat it as a special case, and not return it when listing filesystems. Instead if some module needs to have access to the cache (or a subdir of the cache), it might be better to expose a "GetCache" method on storaged imo. The way the cache is handled should also be an implementation detail of storaged (imo).